### PR TITLE
Add documentation for 3.8

### PR DESCRIPTION
**Pull request by ddfreyne**
Blocked by having 3.8 released.

This also changes the way the rules are explained, because having the examples at the end was rather confusing. I updated it so that the examples are interspersed in the explanation. I also updated some of the markup.

Because of the other changes in this PR, it might be better to read the individual commits. Sorry!

### CODE REVIEW

**Comment by ddfreyne** [comment]
CC @nanoc/contributors

File: content/docs/reference/rules.md [hunk 1]
```diff
@@ -15,215 +15,176 @@ Each item has exactly one compilation rule and one routing rule. Similarly, each
...
 
-To filter an item representation, call `#filter` pass the name of the filter as argument, along with any other filter arguments. For example, the following will call the `:erb` filter on the current item representation:
+    #!ruby
+    compile "/foo/" do
+      # (compilation code here)
+    end
 
-<pre title="Calling the “erb” filter"><code class="language-ruby">
-filter :erb</code></pre>
+The argument for the [`#compile`](/docs/api/Nanoc/CompilerDSL.html#compile-instance_method)  is exactly the same as the argument for `#route`; see the [routing rules](#routing-rules) section above for details.
```
Comment (rosetree) [inline content/docs/reference/rules.md]: > method)  is

Doubled space; I don’t think that was intended.

**Comment by rosetree** [comment]
:+1: looks good! I understood most of it.

**Comment by gpakosz** [comment]
:+1: for the `commands_dirs` description

File: content/docs/reference/rules.md (hunk 1 above)
Comment (ddfreyne) [inline content/docs/reference/rules.md]: Thanks, fixed!

**Comment by bobthecow** [comment]
:+1:

**Pull request merged**